flight/targets/pipxtreme/fw/pios_board.c

```diff
@@ -266,148 +170,13 @@ void PIOS_Board_Init(void) {
-		if (PIOS_RFM22B_Init(&pios_rfm22b_id, PIOS_RFM22_SPI_PORT, rfm22b_cfg->slave_num, rfm22b_cfg)) {
-			PIOS_Assert(0);
-		}
-
-		rfm22bstatus.DeviceID = PIOS_RFM22B_DeviceID(pios_rfm22b_id);
-		rfm22bstatus.BoardRevision = PIOS_RFM22B_ModuleVersion(pios_rfm22b_id);
-
-		// Configure the radio com interface
-		uint8_t *rx_buffer = (uint8_t *)PIOS_malloc(PIOS_COM_RFM22B_RF_RX_BUF_LEN);
-		uint8_t *tx_buffer = (uint8_t *)PIOS_malloc(PIOS_COM_RFM22B_RF_TX_BUF_LEN);
-		PIOS_Assert(rx_buffer);
-		PIOS_Assert(tx_buffer);
-		if (PIOS_COM_Init(&pios_com_rfm22b_id, &pios_rfm22b_com_driver, pios_rfm22b_id,
-					rx_buffer, PIOS_COM_RFM22B_RF_RX_BUF_LEN,
-					tx_buffer, PIOS_COM_RFM22B_RF_TX_BUF_LEN)) {
-			PIOS_Assert(0);
-		}
-
-		// Set the RF data rate on the modem to ~2X the selected buad rate because the modem is half duplex.
-		enum rfm22b_datarate datarate = RFM22_datarate_64000;
-		switch (hwTauLink.MaxRfSpeed) {
-			case HWTAULINK_MAXRFSPEED_9600:
-				datarate = RFM22_datarate_9600;
-				break;
-			case HWTAULINK_MAXRFSPEED_19200:
-				datarate = RFM22_datarate_19200;
-				break;
-			case HWTAULINK_MAXRFSPEED_32000:
-				datarate = RFM22_datarate_32000;
-				break;
-			case HWTAULINK_MAXRFSPEED_64000:
-				datarate = RFM22_datarate_64000;
-				break;
-			case HWTAULINK_MAXRFSPEED_100000:
-				datarate = RFM22_datarate_100000;
-				break;
-			case HWTAULINK_MAXRFSPEED_192000:
-				datarate = RFM22_datarate_192000;
-				break;
-		}
-
-		/* Set the modem Tx poer level */
-		switch (hwTauLink.MaxRfPower) {
-			case HWTAULINK_MAXRFPOWER_125:
-				PIOS_RFM22B_SetTxPower(pios_rfm22b_id, RFM22_tx_pwr_txpow_0);
-				break;
-			case HWTAULINK_MAXRFPOWER_16:
-				PIOS_RFM22B_SetTxPower(pios_rfm22b_id, RFM22_tx_pwr_txpow_1);
-				break;
-			case HWTAULINK_MAXRFPOWER_316:
-				PIOS_RFM22B_SetTxPower(pios_rfm22b_id, RFM22_tx_pwr_txpow_2);
-				break;
-			case HWTAULINK_MAXRFPOWER_63:
-				PIOS_RFM22B_SetTxPower(pios_rfm22b_id, RFM22_tx_pwr_txpow_3);
-				break;
-			case HWTAULINK_MAXRFPOWER_126:
-				PIOS_RFM22B_SetTxPower(pios_rfm22b_id, RFM22_tx_pwr_txpow_4);
-				break;
-			case HWTAULINK_MAXRFPOWER_25:
-				PIOS_RFM22B_SetTxPower(pios_rfm22b_id, RFM22_tx_pwr_txpow_5);
-				break;
-			case HWTAULINK_MAXRFPOWER_50:
-				PIOS_RFM22B_SetTxPower(pios_rfm22b_id, RFM22_tx_pwr_txpow_6);
-				break;
-			case HWTAULINK_MAXRFPOWER_100:
-				PIOS_RFM22B_SetTxPower(pios_rfm22b_id, RFM22_tx_pwr_txpow_7);
-				break;
-			default:
-				// do nothing
-				break;
-		}
-
-		// Set the radio configuration parameters.
-		PIOS_RFM22B_Config(pios_rfm22b_id, datarate, hwTauLink.MinChannel, hwTauLink.MaxChannel, hwTauLink.CoordID, is_oneway, ppm_mode, ppm_only);
-
-		// Reinitilize the modem to affect te changes.
-		PIOS_RFM22B_Reinit(pios_rfm22b_id);
-	} else {
-		rfm22bstatus.LinkState = RFM22BSTATUS_LINKSTATE_DISABLED;
-	}
-
-	// Update the object
-	RFM22BStatusSet(&rfm22bstatus);
+	PIOS_HAL_configure_RFM22B(hwTauLink.Radio, bdinfo->board_type,
```

Comment:
The types in hwshared include all the possibilities and the children can include any subset.  The stuff in pios_port must check for all types and this is enforced by the compiler.  If someone does a switch statement on the child type in the individual hardware module it enforces them to the subset.